### PR TITLE
Add cmdlet to generate ARM templates for Logic App industry link

### PR DESCRIPTION
**Pull request by ongk**
Add cmdlet to generate ARM templates for Logic App industry link. Added and updated documentation on usage. Bug fixes in logic app templates.

Supported data sources:
- Azure Blob Storage
- Azure Event Hubs
- Dataverse

Supported data sinks:
- Dataverse

To be added in separate PR:
- Custom Connector

### CODE REVIEW

File: scripts/modules/MsIndustryLinks/azureDeploymentPackage/azureblobstorage/connection.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
```
Comment (alexandrakoller) [inline scripts/modules/MsIndustryLinks/azureDeploymentPackage/azureblobstorage/connection.json:2]: Is there a reason why we have gone with ARM templates over bicep?

File: scripts/modules/MsIndustryLinks/azureDeploymentPackage/azureblobstorage/connection.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  "contentVersion": "1.0.0.0",
+  "parameters": {
```
Comment (alexandrakoller) [inline scripts/modules/MsIndustryLinks/azureDeploymentPackage/azureblobstorage/connection.json:4]: Is there a template parameters file we could provide?

File: scripts/modules/MsIndustryLinks/azureDeploymentPackage/AzureDeploymentPackage.psm1 [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+    should contain at least one workflow template.
+
+    .Parameter OutputDirectory
+    The directory path where the ARM templates will be saved.
+
+    .Example
+    # Generate ARM templates from Logic App workflow templates
+    New-AzureDeploymentPackage -WorkflowConfigFile workflow.json -TemplateDirectory templates -OutputDirectory output
+#>
+function New-AzureDeploymentPackage {
```
Comment (alexandrakoller) [inline scripts/modules/MsIndustryLinks/azureDeploymentPackage/AzureDeploymentPackage.psm1:29]: Will we be adding in some scripts to automate the deployment of the assets generated in this module?
Without these and documentation, I am finding this hard to set up and test.

**Review by alexandrakoller — COMMENTED** [review]
We should try keep our commits smaller and more frequent to make these larger pull requests easier to review. When we look at coding standards, we should add documentation standards too because there are limited comments that have made understanding some of what is going on in here a bit challenging.
A few questions within the PR before I approve.

File: scripts/modules/MsIndustryLinks/azureDeploymentPackage/AzureDeploymentPackage.psm1 [hunk 4]
```diff
@@ -0,0 +1,329 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
[the remark this answers is not in the document]
Comment (ongk) [reply scripts/modules/MsIndustryLinks/azureDeploymentPackage/AzureDeploymentPackage.psm1:1]: Yes it does. But I didn't want to add more file changes to this PR as this already adds a bunch of new files.

**Comment by ongk** [comment]
> We should try keep our commits smaller and more frequent to make these larger pull requests easier to review. When we look at coding standards, we should add documentation standards too because there are limited comments that have made understanding some of what is going on in here a bit challenging. A few questions within the PR before I approve.

Agreed. But the majority of the changes in this PR is adding a bunch of new template files so that can't be avoided.

File: scripts/modules/MsIndustryLinks/azureDeploymentPackage/azureblobstorage/connection.json (hunk 1 above)
Comment (ongk) [reply scripts/modules/MsIndustryLinks/azureDeploymentPackage/azureblobstorage/connection.json:2]: Bicep files are hard to customize programmatically. In the end, I'd have to compile into ARM template to customize anyways.

File: scripts/modules/MsIndustryLinks/azureDeploymentPackage/azureblobstorage/connection.json (hunk 2 above)
Comment (ongk) [reply scripts/modules/MsIndustryLinks/azureDeploymentPackage/azureblobstorage/connection.json:4]: What is the reason?

File: scripts/modules/MsIndustryLinks/azureDeploymentPackage/AzureDeploymentPackage.psm1 (hunk 3 above)
Comment (ongk) [reply scripts/modules/MsIndustryLinks/azureDeploymentPackage/AzureDeploymentPackage.psm1:29]: Yes, documentation is added in another PR.

File: scripts/modules/MsIndustryLinks/azureDeploymentPackage/azureblobstorage/connection.json (hunk 2 above)
Comment (alexandrakoller) [reply scripts/modules/MsIndustryLinks/azureDeploymentPackage/azureblobstorage/connection.json:4]: I did just realise that when you script the creation of the assets, that it wont be required.

**Review by alexandrakoller — APPROVED** [review]

**Pull request merged**